components/input/Input.js

```diff
@@ -83,10 +83,19 @@ export default class Input extends Component {
     this.handleDecreaseValue = ::this.handleDecreaseValue;
 
     this.state = {
-      value: props.type === 'number' ? Number(props.value) || '' : props.value || '',
+      value: this.parsePropsValue(),
     };
   }
 
+  parsePropsValue() {
+    const { input: { value }, type } = this.props;
```

Comment:
Q: Is this a way to get a deeper key? So you could also do
```js
const { input: { value: { evenDeeperKey } } = this.props;
```
?

Reply:
@InstaK Yes indeed!

Reply:
Cool, I was looking for a way to do this yesterday or last week I remember. Is there still a way to rename them then? So If I wanted to rename `evenDeeperKey`, could I do:

Reply:
In that case you'd have to do something like this:
`const { input: { value: { evenDeeperKey: renamedKey } }, anotherProp } = this.props;`

But this very quickly becomes unreadable :D

Reply:
Fair enough, thanks :)

 :ok: by @InstaK on 27/03/2018 16:14:57